NossaCasaDoCodigo.Testes/TestaCarrinho.cs

```diff
@@ -11,19 +11,50 @@ public class TestaCarrinho
     {
         private static CarrinhoDeCompra Carrinho = new CarrinhoDeCompra();
         [Fact]
-        public void DataHoraDaVendaDeveSerRegistrada()
+        public void DeveTerAoMenosUmLivroParaFecharOCarrinho()
+        {
+            var CarrinhoZerado = new CarrinhoDeCompra();
+            Assert.Throws<ArgumentNullException>(() => CarrinhoZerado.FinalizarCompra());
+        }
+
+        [Fact]
+        public void ValorDeDataHoraDaVendaDeveSerRegistrada()
         {
             Carrinho.AdicionaProduto(LivrosHelpers.LIVRO1, 1);
             Carrinho.FinalizarCompra();
 
             Assert.Equal(DateTime.Today, Carrinho.DataHoraVenda.Date);
         }
 
-
         [Fact]
         public void OLivroDeveSerCadastradoNoEstoqueParaFazerParteDeUmaVenda()
         {
             Assert.Throws<ArgumentNullException>(() => Carrinho.AdicionaProduto(null, 1));
         }
+
+        [Theory]
+        [MemberData(nameof(LivroObjeto))]
+        [MemberData(nameof(LivroObjeto))]
+        public void QuantidadeDeLivrosVendidasDeveSerMaiorQueZero(string titulo, string resumo, string sumario, int paginas,
+            string iSBN, Autor autor, Categoria categoria, double edicao, decimal preco, int quantidade)
+        {
+            Assert.Throws<ArgumentException>(() => Carrinho.AdicionaProduto(new Livro(titulo, resumo, sumario, paginas,
+            iSBN, autor, categoria, edicao, preco), quantidade));
+        }
+
+        public static IEnumerable<object[]> LivroObjeto
+        {
+            get
+            {
+                return new[]
+                {
+                    new object[]{"Titulo Novo", LivrosHelpers.RESUMO_COM_MAIS_DE_500_CARACTERES, "Teste", 20, "978-12-34567-89-0",
+                        new Autor("nome", "email@gmail.com", "minha descrição"),new Categoria("categoria"), 2.1, 50.95, 0 },
+
+                    new object[]{"Titulo Novo", LivrosHelpers.RESUMO_COM_MAIS_DE_500_CARACTERES, "Teste", 20, "978-12-34567-89-0",
```

Comment:
Se você tem a entidade Livro, por que daria new em um object?

Reply:
Então, já tinha feito um teste com isso e refiz para ter certeza. Fiz esse objeto com Livro[], mas na hora de passar o teste com a linha `[MemberData(nameof(TeoriasInvalidasISBN))]` exige um `IEnumerable<object[]>`.
Outro ponto é que todos os códigos de exemplo (documentação, posts e Stack Overflow) sempre utilizam `IEnumerable<object[]>`.

Reply:
Usar object não é tão ruim, se a ferramenta te força a isso, é uma forma dela tentar um polimorfismo da forma mais abstrata possível, mas se ela tiver suporte a algum generic, por que não tentar?

Reply:
Tentei IEnumerable<Livro> e sem sucesso também. A ferramenta quer IEnumerable<Object[]>.